packages/flutter/lib/src/widgets/inherited_theme.dart

```diff
@@ -6,6 +6,38 @@
 library;
 
 import 'framework.dart';
+import 'inherited_model.dart';
+
+/// A generic selector interface for theme data.
+///
+/// This interface allows for selecting specific aspects of theme data without
+/// coupling to specific theme implementations like Material or Cupertino.
+///
+/// The type parameter [T] represents the type of the theme data object
+/// (e.g., `ThemeData`, `CupertinoThemeData`).
+/// The type parameter [V] represents the type of the value being selected
+/// from the theme data (e.g., `Color`, `TextStyle`).
+abstract interface class ThemeSelector<T, V> {
+  /// Creates a [ThemeSelector] from a function.
+  ///
+  /// This factory constructor allows for creating a selector using a lambda syntax.
+  factory ThemeSelector.from(V Function(T) selector) = _FunctionThemeSelector<T, V>;
+
+  /// Selects a value of type [V] from the theme data of type [T].
+  ///
+  /// The actual theme data type is determined by the implementation.
+  V selectFrom(T themeData);
+}
+
+/// A [ThemeSelector] implementation that wraps a function.
+class _FunctionThemeSelector<T, V> implements ThemeSelector<T, V> {
+  const _FunctionThemeSelector(this._selector);
+
+  final V Function(T) _selector;
+
+  @override
+  V selectFrom(T themeData) => _selector(themeData);
```

Comment:
probably better to implement an equal function and  hashcode so that at least the same _selector will equal each other

Reply:
Sure, I will add.